repos/fdbt-site/tests/pages/api/periodValidity.test.ts

```diff
@@ -36,6 +43,43 @@ describe('periodValidity', () => {
         expect(writeHeadMock).toBeCalledWith(302, { Location: '/ticketConfirmation' });
     });
 
+    it('correctly generates product info, and then redirects to /productDetails if update in edit mode', async () => {
+        const { req, res } = getMockRequestAndResponse({
+            cookieValues: {},
+            body: { periodValid: '24hr' },
+            session: {
+                [MATCHING_JSON_ATTRIBUTE]: expectedPeriodGeoZoneTicketWithMultipleProducts,
+                [MATCHING_JSON_META_DATA_ATTRIBUTE]: {
+                    productId: '2',
+                    matchingJsonLink: 'test/path',
+                },
+            },
+            mockWriteHeadFn: writeHeadMock,
+        });
+
+        await periodValidity(req, res);
```

Comment:
I think we also need a test for when productEndTime is in the request, as we have missed a bug here.

Reply:
there will never be productEndTime in the request

Reply:
It looks like there should be, because its an input the user has access to in the form (/selectPeriodValidity has an input as below 
![Screenshot 2022-12-01 at 10 40 47](https://user-images.githubusercontent.com/111580926/205032061-793be2ee-f42a-4fb9-b1ba-a35e3e825627.png)

Im not sure why it doesn't work, as it doesnt ever send the value to the API, but it's irrelevant I think anyway.